site/app/models/LateDaysCalculation.php

```diff
@@ -0,0 +1,291 @@
+                }
+                //If late days used - extensions applied > allowed per assignment then status is "Bad..."
+                if ($curr_late_used - $submissions[$i]['extensions'] > $submissions[$i]['assignment_allowed']) {
+                    $curr_status = "Bad";
+                    $curr_bad_modifier = " too many used for this assignment";
+                    $late_flag = false;
+                }
+                //If late days used - extensions applied > allowed per term then status is "Bad..."
+                if ($curr_late_used - $submissions[$i]['extensions'] >  $curr_allowed_term - $total_late_used) {
+                    $curr_status = "Bad";
+                    $curr_bad_modifier = " too many used this term";
+                    $late_flag = false;
+                }
+                //A submission cannot be late and bad simultaneously. If it's late calculate late days charged. Cannot
+                //be less than 0 in cases of excess extensions. Decrement remaining late days.
+                if ($late_flag) {
+                    $curr_late_charged = $curr_late_used - $submissions[$i]['extensions'];
+                    $curr_late_charged = ($curr_late_charged < 0) ? 0 : $curr_late_charged;
+                    $total_late_used += $curr_late_charged;
+                }
+
+                $curr_remaining_late = $curr_allowed_term - $total_late_used;
+                $curr_remaining_late = ($curr_remaining_late < 0) ? 0 : $curr_remaining_late;
+
+                $submission_latedays['user_id'] = $submissions[$i]['user_id'];
+                $submission_latedays['g_title'] = $submissions[$i]['g_title'];
+                $submission_latedays['allowed_per_term'] = $curr_allowed_term;
+                $submission_latedays['allowed_per_assignment'] = $submissions[$i]['assignment_allowed'];
+                $submission_latedays['late_days_used'] = $curr_late_used;
+                $submission_latedays['extensions'] = $submissions[$i]['extensions'];
+                $submission_latedays['status'] = $curr_status.$curr_bad_modifier;
+                $submission_latedays['late_days_charged'] = $curr_late_charged;
+                $submission_latedays['remaining_days'] = $curr_remaining_late;
+                $submission_latedays['total_late_used'] = $total_late_used;
+                $submission_latedays['eg_submission_due_date'] = $submissions[$i]['eg_submission_due_date'];
+
+                $late_day_usage[$submissions[$i]['g_id']] = $submission_latedays;
+            }
+
+            $all_latedays[$student['user_id']] = $late_day_usage;
+
+        }
+
+        return $all_latedays;
+    }
+    
+     /**
+     * For the given user id generate the late day usage HTML table.
+     * @param $user_id String. The user id of the user whose table you want.
+     * @return string. The string representation of the HTML table.
+     */
+    public function generate_table_for_user($user_id){
```

Comment:
I don't use this function or the one below it, but they are used to create the student info panel in the TA Grading page, so I added them to be used when James needs that info for the TA Grading page port. We will probably want to move those functions to the TA Grading view when we get rid of this class and move it's info into the Gradeable model